spec/fixtures/files/test_api.json

```diff
@@ -131,6 +134,49 @@
         ]
       }
     },
+    "RecursiveHat": {
+     "id": "RecursiveHat",
+     "type": "object",
+     "properties": {
+      "bool": {
+       "type": "boolean",
+       "description": "The bool value."
+      },
+      "double": {
+       "type": "number",
+       "description": "The double value.",
+       "format": "double"
+      },
+      "image": {
+       "$ref": "Image",
+       "description": "The image value."
+      },
+      "int": {
+       "type": "integer",
+       "description": "The int value.",
+       "format": "int32"
+      },
+      "kind": {
+       "type": "string",
+       "description": "The kind, fixed to \"test#recursiveHat\".",
+       "default": "test#recursiveHat"
+      },
+      "list": {
+       "type": "array",
+       "description": "The list of typed values.",
+       "items": {
+        "$ref": "RecursiveHat"
+       }
+      },
+      "string": {
+       "type": "string",
+       "description": "The string value."
+      }
+    },
+   "additionalProperties": {
+    "$ref": "RecursiveHat"
+   }
+  },
```

Comment:
Something is weird about the indenting here.

Reply:
Fixed in the latest commit.